### PR TITLE
search panel is added to index

### PR DESCRIPTION
**Pull request by mainstreetswagger**
added search panel.

### CODE REVIEW

**Comment by mainstreetswagger** [comment]
how to apply one definite migration to update db? 
need something command like update-database <migration file name>

File: UtopiaCity/Services/Life/LifeService.cs [hunk 1]
```diff
@@ -70,6 +70,55 @@ public void Remove(Event ev)
...
-        }        
+        }
+        public List<Event> Search(EventDto dto)
+        {
+            var list = SearchByDates(from: dto.From, to: dto.To);
+            list = SearchByEventType(list, dto.EventType);
+            list = SearchByWord(list, dto.SearchWord);
+            return list;
+        }
+        private List<Event> SearchByDates(DateTime? from, DateTime? to)
```
Comment (Ibraga96) [inline UtopiaCity/Services/Life/LifeService.cs]: Why you decided to use Private methods in service class?

File: UtopiaCity/wwwroot/js/life_search.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    function onReceiving() {
+        let res = JSON.parse(xhttp.responseText);
+        let table = document.getElementById('table');
+
+        let length = parseInt(table.rows.length);
+        for (let i = 0; i < length; i++) {
+            table.deleteRow(0);
+        }
+
```
Comment (Ibraga96) [inline UtopiaCity/wwwroot/js/life_search.js]: can't say for javascript , but looks interesting

File: UtopiaCity/Data/ApplicationDbContext.cs [hunk 3]
```diff
@@ -21,11 +21,7 @@ public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options)
         public DbSet<WeatherReport> WeatherReports { get; set; }
         public DbSet<ForPassenger> ForPassengers { get; set; }
         public DbSet<TransportManager> TransportManagers { get; set; }
```
Comment (crocomoth) [inline UtopiaCity/Data/ApplicationDbContext.cs:24]: Not your code?

File: UtopiaCity/Services/Life/LifeService.cs (hunk 1 above)
Comment (crocomoth) [reply UtopiaCity/Services/Life/LifeService.cs]: Seems like it is internal methods, only to be used by service itself, it is okay

File: UtopiaCity/Services/Life/LifeService.cs [hunk 4]
```diff
@@ -70,6 +70,55 @@ public void Remove(Event ev)
         public bool Exist(string id)
         {
             return _dbContext.Events.Any(x => x.Id.Equals(id));
-        }        
+        }
+        public List<Event> Search(EventDto dto)
+        {
+            var list = SearchByDates(from: dto.From, to: dto.To);
```
Comment (crocomoth) [inline UtopiaCity/Services/Life/LifeService.cs:54]: I dont think that this search is correct way to do that, but okay

File: UtopiaCity/Views/Life/_SearchPanel.cshtml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    border-style:solid;
+    border-width:thin;
+    border-color:blue;
+    padding: 15px;
+    margin:5px;">
+    <form asp-controller = "Life" asp-action = "Search">
+        <div class="row">
+            <div class="form-group col-sm">
+                <label>Searching Word</label><br />
+                <input type="text" name="SearchWord" value="" style="border-radius: 10px" id="sword" />
```
Comment (crocomoth) [inline UtopiaCity/Views/Life/_SearchPanel.cshtml]: maybe 'border-radius' should instead be class in css, and applied if needed

File: UtopiaCity/wwwroot/js/life_search.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        for (let i = 0; i < res.length; i++) {
+            let row = table.insertRow(i);
+            let cell1 = row.insertCell(0);
+            let cell2 = row.insertCell(1);
+            let cell3 = row.insertCell(2);
+            let cell4 = row.insertCell(3);
+            let cell5 = row.insertCell(4);
+            console.log(res[i].description);
+            cell1 = res[i].date;
```
Comment (crocomoth) [inline UtopiaCity/wwwroot/js/life_search.js]: I think this is a rather strange way to do it, but it is nice that you are going extra mile

**Review by crocomoth — CHANGES_REQUESTED** [review]

File: UtopiaCity/wwwroot/js/life_search.js (hunk 2 above)
Comment (mainstreetswagger) [reply UtopiaCity/wwwroot/js/life_search.js]: Could be better if worked.

File: UtopiaCity/Data/ApplicationDbContext.cs (hunk 3 above)
Comment (mainstreetswagger) [reply UtopiaCity/Data/ApplicationDbContext.cs:24]: No it doesn't refer to me.

File: UtopiaCity/Services/Life/LifeService.cs (hunk 1 above)
Comment (mainstreetswagger) [reply UtopiaCity/Services/Life/LifeService.cs]: Yes, for now i use them only in service itself.

File: UtopiaCity/Data/ApplicationDbContext.cs (hunk 3 above)
Comment (mainstreetswagger) [reply UtopiaCity/Data/ApplicationDbContext.cs:24]: Well, when i merged, i got dublicate dbsets. I had to delete them.

**Review by crocomoth — APPROVED** [review]

**Review by AnatoliyDubovtsev — APPROVED** [review]

**Pull request closed without merging**